File: src/main/java/model/Fruit.java
package model;

public class Fruit {

    private double price;
    private FruitName name;

    public Fruit(FruitName name, double price) {
        this.price = price;
        this.name = name;
    }

    public double getPrice() {
        return price;
    }

    public String getName() {
        return name.getFruitType();
    }
}

File: src/main/java/model/Fruit.kt
package model

class Fruit {
    private var price = 0.0
    private var name: FruitName? = null

    fun Fruit(name: FruitName?, price: Double) {
        this.price = price
        this.name = name
    }

    fun getPrice(): Double {
        return price
    }

    fun getName(): String? {
        return name!!.type
    }
}